contrib/lib/src/templates/mod.rs

```diff
@@ -6,6 +6,7 @@ extern crate glob;
 #[cfg(feature = "handlebars_templates")] mod handlebars_templates;
 mod engine;
 mod context;
+pub mod template_engine;
```

Comment:
Unfortunately, it has to be. Otherwise, compilation fails with

```
error[E0603]: module `template_engine` is private
  --> contrib/lib/src/lib.rs:77:40
   |
77 | pub use templates::{Template, Engines, template_engine::TemplateEngine};```

Reply:
You want `mod metadata` here (assuming you rename the `template_engine.rs` file to `metedata.rs`) and a `pub use self::metadata::TemateMetadata'` below.